storagedriver/azure/azure.go

```diff
@@ -99,26 +103,34 @@ func (d *Driver) GetContent(path string) ([]byte, error) {
 }
 
 // PutContent stores the []byte content at a location designated by "path".
-func (d *Driver) PutContent(path string, contents []byte) error {
+func (d *driver) PutContent(path string, contents []byte) error {
 	return d.client.PutBlockBlob(d.container, path, ioutil.NopCloser(bytes.NewReader(contents)))
 }
 
 // ReadStream retrieves an io.ReadCloser for the content stored at "path" with a
 // given byte offset.
-func (d *Driver) ReadStream(path string, offset int64) (io.ReadCloser, error) {
+func (d *driver) ReadStream(path string, offset int64) (io.ReadCloser, error) {
+	if offset < 0 {
+		return nil, storagedriver.InvalidOffsetError{Path: path, Offset: offset}
+	}
+
 	if ok, err := d.client.BlobExists(d.container, path); err != nil {
 		return nil, err
 	} else if !ok {
 		return nil, storagedriver.PathNotFoundError{Path: path}
 	}
 
-	size, err := d.CurrentSize(path)
+	info, err := d.client.GetBlobProperties(d.container, path)
 	if err != nil {
 		return nil, err
 	}
 
-	if offset >= int64(size) {
+	size := int64(info.ContentLength)
+	if offset < 0 || offset > size {
```

Comment:
Offset larger than size is not an invalid offset. Offset is only invalid if it is less than 0. Offset larger than size should return an `io.EOF` reader. Maybe this could be added to `Base`.

The reason for this is that an offset isn't really valid or invalid until the data is read. For example, you could stat a file, then call this method, thinking that the size is valid. By the time the file is read, that size is no longer valid.